src/security/PeerOptions.cc

```diff
@@ -512,7 +512,7 @@ Security::PeerOptions::parseOptions()
 
     }
 
-#if SSL_OP_NO_SSLv2
+#if defined(SSL_OP_NO_SSLv2)
     // compliance with RFC 6176: Prohibiting Secure Sockets Layer (SSL) Version 2.0
     op = op | SSL_OP_NO_SSLv2;
```

Comment:
Similar preprocessor change concern here: After this PR change, this "Prohibiting Secure Sockets Layer" code prohibits nothing when SSL_OP_NO_SSLv2 is zero, misleading the reader. Please clarify why this preprocessor condition should be changed. Referring to another change request explanation, it it applies, is fine, of course.

Reply:
> OpenSSLv3 makes the definition of all options an expression instead of a constant value

OpenSSL v3 defines `SSL_OP_NO_SSLv2` to be `0x0`. `0x0` is a constant value AFAICT. Moreover, `SSL_OP_NO_SSLv2` was already set to zero (using a different spelling) in OpenSS v1.1!

> it is either desirable, or at worst harmelss, to set this option whenever the macro is #defined.

When `SSL_OP_NO_SSLv2` is defined to a falsy value `0x0` (as it is in OpenSSL v3), it is _not_ desirable to "set this option" (IMO) because
* doing so misleads the reader that into thinking that this "Prohibiting Secure Sockets Layer v2" code prohibits something when, in fact, it does not;
* it increases the PR footprint without any runtime or code reading time benefit (that I can see).

In other words, the official code is better than the proposed PR change. Please revert this and similar changes or explain why they are needed in this PR.

Reply:
I also checked other tagged OpenSSL releases (all I 200+ could find), and, AFAICT, all those that define `SSL_OP_NO_SSLv2` macro set it to a constant value (`0x0`, `0x00000000U`, or `0x01000000L`).

Reply:
Based on the suggestion in the other change request:

```suggestion
    if (SSL_OP_NO_SSLv2)
        op = op | SSL_OP_NO_SSLv2;
```

Yes, the added `if` does not have an effect on `op` final value, but it improves consistency of treating `SSL_OP_X` values and highlights an important caveat: being `#defined` is not enough for `SSL_OP_X` to be supported. If you prefer to be more explicit about it, we can do this:

```suggestion
    if (SSL_OP_NO_SSLv2) // may be #defined but not supported
        op = op | SSL_OP_NO_SSLv2;
```

Reply:
Done in [3815b94](https://github.com/squid-cache/squid/pull/694/commits/3815b94d14858c503ad67adcaec8d98b1c1d5b24)